openedx/enterprise-subsidy
# [Do not merge] Segfault example event bus crash

### Description
This PR is an example setup of me attempting to add edx-event-bus-kafka support to this repo. When I attempt to run the new management command I receive the following:
```
(.python) ➜  enterprise-subsidy git:(bbeggs/kafka) ✗ python -X faulthandler manage.py consume_events -t dev.course-enrollment-created -g my_group -s org.openedx.learning.course.enrollment.created.v1
2023-04-12 17:44:00,472 INFO 72259 [edx_event_bus_kafka.internal.consumer] [user None] [ip None] consumer.py:254 - Running consumer for {'full_topic': 'dev-dev.course-enrollment-created', 'consumer_group': 'my_group', 'expected_signal': <OpenEdxPublicSignal: org.openedx.learning.course.enrollment.created.v1>}
Fatal Python error: Segmentation fault

Current thread 0x00000001f5b11b40 (most recent call first):
  File "/Users/bbeggs/workspace/enterprise-subsidy/.python/lib/python3.8/site-packages/confluent_kafka/deserializing_consumer.py", line 98 in poll
  File "/Users/bbeggs/workspace/enterprise-subsidy/.python/lib/python3.8/site-packages/edx_event_bus_kafka/internal/consumer.py", line 274 in _consume_indefinitely
  File "/Users/bbeggs/workspace/enterprise-subsidy/.python/lib/python3.8/site-packages/edx_event_bus_kafka/internal/consumer.py", line 319 in consume_indefinitely
  File "/Users/bbeggs/workspace/enterprise-subsidy/.python/lib/python3.8/site-packages/edx_event_bus_kafka/internal/consumer.py", line 647 in handle
  File "/Users/bbeggs/workspace/enterprise-subsidy/.python/lib/python3.8/site-packages/django/core/management/base.py", line 398 in execute
  File "/Users/bbeggs/workspace/enterprise-subsidy/.python/lib/python3.8/site-packages/django/core/management/base.py", line 354 in run_from_argv
  File "/Users/bbeggs/workspace/enterprise-subsidy/.python/lib/python3.8/site-packages/django/core/management/__init__.py", line 413 in execute
  File "/Users/bbeggs/workspace/enterprise-subsidy/.python/lib/python3.8/site-packages/django/core/management/__init__.py", line 419 in execute_from_command_line
  File "manage.py", line 29 in <module>
[1]    72259 segmentation fault  python -X faulthandler manage.py consume_events -t  -g my_group -s 

```

### Testing instructions

1. Checkout this branch
2. From your devstack run `make dev.up.kafka-control-center`
3. Ensure your kafka server url/port coniguration are correctly set in enterprise_subsidy/settings/base.py
4. run `python -X faulthandler manage.py consume_events -t dev.course-enrollment-created -g my_group -s org.openedx.learning.course.enrollment.created.v1`

I cobbled together this toy python script to send a message and consume it. I am able to run this script successfully without a segfault. 


```
(.python) ➜  enterprise-subsidy git:(bbeggs/kafka) ✗ python -V
Python 3.8.13
(.python) ➜  enterprise-subsidy git:(bbeggs/kafka) ✗ python -X faulthandler test_kafka.py
```

test_kafka.py
```
from confluent_kafka import Producer, Consumer, KafkaError, KafkaException
import socket
from inspect import getmembers

conf = {
        'bootstrap.servers': "localhost:9092",
        'client.id': socket.gethostname(),
        'security.protocol': 'PLAINTEXT'

}

producer = Producer(conf)
producer.produce("test", key="key", value="value")
print("About to flush")
producer.flush()
print("done")


consumer_conf = {
        'bootstrap.servers': "localhost:9092",
        'client.id': socket.gethostname(),
        'security.protocol': 'PLAINTEXT',
        'group.id': "foo",
        'auto.offset.reset': 'smallest'
}

consumer = Consumer(consumer_conf)
running = True

def basic_consume_loop(consumer, topics):
    try:
        consumer.subscribe(topics)

        while running:
            msg = consumer.poll(timeout=1.0)
            if msg is None:
                    continue

            if msg.error():
                if msg.error().code() == KafkaError._PARTITION_EOF:
                    # End of partition event
                    sys.stderr.write('%% %s [%d] reached end at offset %d\n' %
                                     (msg.topic(), msg.partition(), msg.offset()))
                elif msg.error():
                    raise KafkaException(msg.error())
            else:
                print(getmembers(msg))
    finally:
        # Close down consumer to commit final offsets.
        consumer.close()

def shutdown():
    running = False

basic_consume_loop(consumer, ["test"])

```
